lib/modules/manager/poetry/artifacts.ts

```diff
@@ -53,18 +49,16 @@ export function getPoetryRequirement(
     logger.debug('Using poetry version from poetry.lock header');
     return poetryVersionMatch[1];
   }
-  try {
-    const data = parse(existingLockFileContent) as PoetryLock;
-    const lockVersion = data?.metadata?.['lock-version'];
-    if (is.string(lockVersion)) {
-      if (poetryConstraint[lockVersion]) {
-        logger.debug('Using poetry version from poetry.lock metadata');
-        return poetryConstraint[lockVersion];
-      }
-    }
-  } catch (err) {
-    // Do nothing
+
+  const { val: lockVersion } = Result.parse(
+    Lockfile.transform(({ lockVersion }) => lockVersion),
+    existingLockFileContent
+  ).unwrap();
+  if (lockVersion) {
```

Comment:
you lost the second condition, please re-add

Reply:
It's handled during parsing at the schema level